tests/test_unit_spec.py

```diff
@@ -142,7 +166,18 @@ def test_resolve_references(self):
             single_ecu_jobs=[],
             diag_comm_refs=[],
             diag_data_dictionary_spec=DiagDataDictionarySpec(data_object_props=[dop],
-                                                             unit_spec=UnitSpec(units=[unit])
+                                                             unit_spec=UnitSpec(units=[unit],
+                                                                                physical_dimensions=None,
+                                                                                unit_groups=None,
+                                                                                sdgs=[],),
+                                                             dtc_dops = NamedItemList(short_name_as_id, []),
```

Comment:
Do we still need `short_name_as_id` ?

Reply:
yes, because short names may start with a digit, which is not allowed in python. maybe that case should be made easier, e.g. by introducing a `ShortNameItemList` derived from `NamedItemList`? (in this particular instance, one could also just pass through the short names, because the unit test does not (yet) feature one which starts with a digit...)

Reply:
I think they can start with a digit, I have just looked in the spec yesterday and implemented a checker function according to the ISO: https://github.com/mercedes-benz/odxtools/blob/b14262fd4d9b51ad79e40148e9ce663c36ab2387/odxtools/utils.py#L57

Reply:
> Short names can't start with digit, it's the IDs that can

wasn't it you who had a PDX file that committed that atrocity ;)? I'm 100% sure that the original code used the short_name directly.

Reply:
yes, it was an odx from me, and it was in the comparam id, now comparam uses short names for keys

Reply:
the issue with the IDs is, although they have to start with a letter, they may contain hyphens. these are not allowed in short names, but -- in order to settle the balance -- they may start with a digit. IIRC you reported both of these issues...

Reply:
I will recheck this later, either way it's a suggestion that's outside the scope of the PR